unlock-app/src/components/interface/checkout/alpha/Checkout/Password.tsx

```diff
@@ -0,0 +1,100 @@
+import { CheckoutService } from './checkoutMachine'
+import { Connected } from '../Connected'
+import { Button, Input } from '@unlock-protocol/ui'
+import { Fragment } from 'react'
+import { ToastHelper } from '~/components/helpers/toast.helper'
+import { useActor } from '@xstate/react'
+import { PoweredByUnlock } from '../PoweredByUnlock'
+import { Stepper } from '../Stepper'
+import { useCheckoutSteps } from './useCheckoutItems'
+import { ethers } from 'ethers'
+import { useForm } from 'react-hook-form'
+interface Props {
+  injectedProvider: unknown
+  checkoutService: CheckoutService
+}
+
+interface FormData {
+  password: string
+}
+
+export function Password({ injectedProvider, checkoutService }: Props) {
+  const [state, send] = useActor(checkoutService)
+  const { recipients } = state.context
+  const {
+    register,
+    handleSubmit,
+    formState: { isSubmitting, errors },
+  } = useForm<FormData>()
+
+  const onSubmit = async (formData: FormData) => {
+    try {
+      const { password } = formData
+      if (!(password && recipients.length)) {
+        return
+      }
+      const encoded = ethers.utils.defaultAbiCoder.encode(
+        ['bytes32'],
+        [ethers.utils.id(password)]
+      )
+      const privateKey = ethers.utils.keccak256(encoded)
+      const privateKeyAccount = new ethers.Wallet(privateKey)
+      const data = await Promise.all(
+        recipients.map((address) => {
+          const messageHash = ethers.utils.solidityKeccak256(
+            ['string'],
+            [address.toLowerCase()]
+          )
+          const messageHashBinary = ethers.utils.arrayify(messageHash)
+          return privateKeyAccount.signMessage(messageHashBinary)
+        })
+      )
+      send({
+        type: 'SUBMIT_PASSWORD',
+        data,
+      })
+    } catch (error: any) {
+      ToastHelper.error(error?.message)
+    }
+  }
+
+  const stepItems = useCheckoutSteps(checkoutService)
+
+  return (
+    <Fragment>
+      <Stepper position={6} service={checkoutService} items={stepItems} />
+      <main className="h-full px-6 py-2 overflow-auto">
+        <form id="password" className="space-y-4">
+          <Input
+            label="Enter password"
```

Comment:
Let's make it of `type` `password`... and ideally have a way to "reveal" it? wdyt?

Reply:
Maybe that's inside the `Input` component in the UI package?

Reply:
@searchableguy Please set this input to be of type `password`